### PR TITLE
#1576 Positions Lab: Should not require at least 2-3 shapes to be selected

### PR DESCRIPTION
**Pull request by junkiattan**
#1576 

I have included default align functionality for when one shape is selected for the following

1. Align-Left
2. Align-Right
3. Align-Top
4. Align-Bottom
5. Align-Center
6. Align-Center (Horizontal)
7. Align-Center (Vertical)

FTs and UTs returned no error

### CODE REVIEW

File: PowerPointLabs/PowerPointLabs/PositionsLab/PositionsLabMain.cs [hunk 1]
```diff
@@ -96,7 +96,8 @@ public static void AlignLeft(ShapeRange toAlign)
                 case PositionsLabSettings.AlignReferenceObject.SelectedShape:
                     if (toAlign.Count < 2)
                     {
-                        throw new Exception(PositionsLabText.ErrorFewerThanTwoSelection);
+                        // Insert default align left behaviour here
```
Comment (Skaty) [inline PowerPointLabs/PowerPointLabs/PositionsLab/PositionsLabMain.cs]: There's reason to believe that the comments are quite useless, as we know that the `Align` method is the built-in PowerPoint align.

File: PowerPointLabs/PowerPointLabs/PositionsLab/PositionsLabMain.cs [hunk 2]
```diff
@@ -361,7 +367,9 @@ public static void AlignCenter(ShapeRange toAlign, float slideHeight, float slid
                 case PositionsLabSettings.AlignReferenceObject.SelectedShape:
                     if (toAlign.Count < 2)
                     {
-                        throw new Exception(PositionsLabText.ErrorFewerThanTwoSelection);
+                        // Insert default align center behaviour here
+                        toAlign.Align(MsoAlignCmd.msoAlignMiddles, MsoTriState.msoTrue);
+                        toAlign.Align(MsoAlignCmd.msoAlignCenters, MsoTriState.msoTrue);
```
Comment (Skaty) [inline PowerPointLabs/PowerPointLabs/PositionsLab/PositionsLabMain.cs]: Question: why the need to align twice?

File: PowerPointLabs/PowerPointLabs/PositionsLab/PositionsLabMain.cs (hunk 1 above)
Comment (junkiattan) [reply PowerPointLabs/PowerPointLabs/PositionsLab/PositionsLabMain.cs]: @Skaty 

Agreed. It was for the purposes of identifying the point of adding the code while I was working on this. Ill remove them

File: PowerPointLabs/PowerPointLabs/PositionsLab/PositionsLabMain.cs (hunk 2 above)
Comment (junkiattan) [reply PowerPointLabs/PowerPointLabs/PositionsLab/PositionsLabMain.cs]: In order to align center, you need to first align horizontally and then vertically

**Comment by junkiattan** [comment]
@Skaty 

Thanx for the highlighting a potential problem with the removal of exception! I will look into the issue;)

**Comment by junkiattan** [comment]
@Skaty 

Added relevant UTs to check for aligning functionality for one shape. Ran them and they all pass.

File: PowerPointLabs/Test/UnitTest/PositionsLab/PositionsLabAlignTest.cs [hunk 3]
```diff
@@ -433,6 +433,117 @@ public void TestAlignOneCenterDefault()
             CheckShapes(expectedShapes, actualShapes);
         }
 
+        [TestMethod]
+        [TestCategory("UT")]
+        public void TestAlignOneLeftSingleShape()
```
Comment (Skaty) [inline PowerPointLabs/Test/UnitTest/PositionsLab/PositionsLabAlignTest.cs]: What's the difference between this and the method `TestAlignOneLeftDefault`? Same with the other methods.

Comment (junkiattan) [reply PowerPointLabs/Test/UnitTest/PositionsLab/PositionsLabAlignTest.cs]: The reference point is different. For the new tests, the reference points are the respective selected shapes.

Comment (Skaty) [reply PowerPointLabs/Test/UnitTest/PositionsLab/PositionsLabAlignTest.cs]: Thanks for the clarification, didn't spot that bit! :)

**Comment by Skaty** [comment]
@leeyh20 Retest required, since new UTs are added.

**Comment by leeyh20** [comment]
Retested on PPT 2010 with the new UTs. The tests all pass.

**Pull request merged**